### PR TITLE
[IT-1674] Collaborator role for INCLUDE project

### PR DESCRIPTION
**Pull request by zaro0508**
Setup a Collaborator role to access INCLUDE DCC project resources
at Children’s Hospital of Philadelphia (CHOP).

The policy for this role is from the AWS VPC documention[1]
in section `Restricting access to buckets in a specific AWS account`

[1] https://docs.aws.amazon.com/vpc/latest/privatelink/vpc-endpoints-s3.html#vpc-endpoints-policies-s3

### CODE REVIEW

File: org-formation/700-aws-sso/_tasks.yaml [hunk 1]
```diff
@@ -979,3 +983,41 @@ SsoCnbAdmin:
...
+            "Principal": "*",
+            "Action": [
+                "s3:GetObject*",
+                "s3:PutObject*",
+                "s3:DeleteObject*"
+            ],
+            "Resource": "arn:aws:s3:::*",
+            "Condition": {
+              "StringNotLike": {
+                "s3:ResourceAccount": "<CHOP AWS Account ID>",
```
Comment (zaro0508) [inline org-formation/700-aws-sso/_tasks.yaml:1018]: @BrunoGrandePhD could you please provide the CHOP AWS account number?

Comment (BrunoGrandePhD) [reply org-formation/700-aws-sso/_tasks.yaml:1018]: You might want to create three statements, one for each account. I'll review the PR once the account numbers are added.

Comment (zaro0508) [reply org-formation/700-aws-sso/_tasks.yaml]: done

File: org-formation/700-aws-sso/_tasks.yaml [hunk 2]
```diff
@@ -979,3 +983,73 @@ SsoCnbAdmin:
...
+    permissionSetName: 'IncludeDccCollab'
+    sessionDuration: 'PT12H'
+    # policy derived from https://docs.aws.amazon.com/vpc/latest/privatelink/vpc-endpoints-s3.html#vpc-endpoints-policies-s3
+    # Restricting access to buckets in a specific AWS account
+    inlinePolicy: >-
+      {
+        "Version": "2012-10-17",
+        "Statement": [
+          {
+            "Effect": "Deny",
```
Comment (BrunoGrandePhD) [inline org-formation/700-aws-sso/_tasks.yaml:1010]: According to these [docs](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_evaluation-logic-cross-account.html), these `Deny` statements won't be sufficient to grant cross-account access. I think the role also needs explicit `Allow` statements for S3 actions.

File: org-formation/700-aws-sso/_tasks.yaml [hunk 3]
```diff
@@ -979,3 +983,73 @@ SsoCnbAdmin:
...
     permissionSetArn: !CopyValue [ !Sub '${resourcePrefix}-${appName}-admin-permission-set-arn' ]
+
+SsoScicompIncludeDccCollab:
+  Type: update-stacks
+  Template: https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra/v0.2.11/templates/SSO/aws-sso.yaml
+  StackName: !Sub '${resourcePrefix}-${appName}-scicomp-include-dcc-collab'
+  StackDescription: 'SSO: role to access INCLUDE DCC project resources at CHOP'
+  DefaultOrganizationBindingRegion: !Ref primaryRegion
+  DefaultOrganizationBinding:
+    IncludeMasterAccount: true
```
Comment (BrunoGrandePhD) [inline org-formation/700-aws-sso/_tasks.yaml:994]: Do we want this to be `true`?

**Review by BrunoGrandePhD — CHANGES_REQUESTED** [review]

File: org-formation/700-aws-sso/_tasks.yaml (hunk 2 above)
Comment (zaro0508) [reply org-formation/700-aws-sso/_tasks.yaml:1011]: good catch.  I've corrected the policy.  It now provide S3 full access to this role with an explicit S3 deny on all accounts except for the CHOP accounts.

File: org-formation/700-aws-sso/_tasks.yaml [hunk 4]
```diff
@@ -979,3 +983,73 @@ SsoCnbAdmin:
...
+    sessionDuration: 'PT12H'
+    # policy derived from https://docs.aws.amazon.com/vpc/latest/privatelink/vpc-endpoints-s3.html#vpc-endpoints-policies-s3
+    # Restricting access to buckets in a specific AWS account
+    inlinePolicy: >-
+      {
+        "Version": "2012-10-17",
+        "Statement": [
+          {
+            "Effect": "Deny",
+            "Principal": "*",
```
[the remark this answers is not in the document]
Comment (zaro0508) [reply org-formation/700-aws-sso/_tasks.yaml]: done

**Review by BrunoGrandePhD — APPROVED** [review]
As far as I can tell, this role will work. If you want, I can provide a list of S3 actions that we can deny in a second statement to be extra safe for our collaborators (as per the suggestion of the AWS consultant). Things like `s3:CreateBucket` and `s3:DeleteBucket`. 

@ahayden: What do you think?

**Review by ahayden — APPROVED** [review]
This should meet the request in IT-1674. If @BrunoGrandePhD validates that it works for the collab, I will validate that it works internally.

**Comment by BrunoGrandePhD** [comment]
I can't help validate the role until I have access, but I shouldn't have access until you've validated that I can't touch the `scicomp` buckets.

**Pull request merged**